oop1-construct.js

```diff
@@ -20,58 +66,99 @@ Create a class that can categorize at least 6 of the creatures that you recall o
+        let triangleHeight = 2 * (triangleArea / this.sides[0]);
+        this.height = triangleHeight;
+        this.base = this.sides[0];
+        this.area = triangleArea;
 
-const triangle = {
-    name : "triangle",
-    sides : [4,7,7], //lengths of each side
-    base : null,
-    height : null, //determine this height using basic geometry.  You may need to research this one.
-    calcArea : function() {
-        console.log(`${this.name}'s area is calculated to be : ${.5 * this.base * this.height}`);
-    },
-    calcPerimeter : function() {
-        console.log(`${this.name}'s perimeter is calculated to be : ${(this.base) + (this.sides[1] + this.sides[2]) }`)
     }
 }
 
-triangle.base = triangle.sides[0];  //base is the 1st side in the triangle.sides array.
-triangle.height = 6.71; //determine the height using basic geometry.  How do you calculate the height of a triangle with 2 equal sides?  If you use the formula to caluclate the height, this is a bonus.  If you hard code the correct value here, that will be sufficient, but no bonus :(
+triangle = new Triangle("triangle", [4, 7, 7], null, null);
 
-console.log(triangle);
-triangle.calcArea();
-triangle.calcPerimeter();
+triangle.HeronsCalc();
 
-const rectangle = {
-    name : "rectangle",
-    sides : 4,
-    length : 2,
-    width : 5,
-    calcArea : function() {
-        console.log(`${this.name}'s area is calculated to be : ${this.length * this.width}`);
-    },
-    calcPerimeter : function() {
-        console.log(`${this.name}'s perimeter is calculated to be : ${(2 * this.length) + (2 * this.width) }`)
+
+shape = new Shape("triangle", [4, 7, 7], 8, 8);
+console.log(`The triangles area is ${triangle.area}`);
+console.log(`The triangles heigh is ${triangle.height}`);
+
+class Rectangle extends Shape {
+    constructor(name, sides, base, height, area, length, width) {
+        super(name, sides, base, height, area, length, width);
     }
+    calcArea() {
+        let area = this.length * this.width;
+
+
+        this.area = area;
+
+    }
+    calcPerimeter() {
+        console.log(`The perimeter is ${(2 * this.length) + (2 * this.width)}`)
+    }
+
 }
+let rectangle = new Rectangle("rectangle", 4, null, null, null, 2, 5);
 
-console.log(rectangle);
 rectangle.calcArea();
 rectangle.calcPerimeter();
+console.log(`The area of the rectangle is ${rectangle.area}`);
 
-const circle = {
-    name : "circle",
-    sides : 1,
-    radius : 5,
-    calcArea : function() {
-        console.log(`${this.name}'s area is calculated to be : ${(Math.PI * this.radius**2).toFixed(2)}`);
-    },
-    calcCircumference : function() {
+class Circle extends Shape {
+    constructor(name, sides, base, height, area, length, width, radius) {
+        super(name, sides, base, height, area, length, width);
+        this.radius = radius;
+    }
+    calcArea() {
+        console.log(`${this.name}'s area is calculated to be : ${(Math.PI * this.radius ** 2).toFixed(2)}`);
+    }
```

Comment:
Don't forget to save the area, then you can use it in the console.log. It's generally bad practice to do calculations in an output of any kind. Harder to make safe that way. For example, what if it ends up being a negative value? That can't be right. What if somehow due to the input it tries to divide by 0? Exception! Crash! Instead, do the calculation in the first line `calcArea() { this.area = (Math.PI * this.radius ** 2).toFixed(2) }` where you can control it before sending it to the output`